### PR TITLE
Refresh dbus objects based on a signal from host

### PR DESCRIPTION
**Pull request by manojkiraneda**
PHYP sends a sensor event on PLDM_STATE_SET_VERSION
state set to indicate that there is a change in the
mex io enclosure version.

This PR would react to the event and refresh the dbus
objects hosted by PLDM. This PR would also remove extra
unwanted traces.

Fixes : SW550806

Signed-off-by: Manojkiran Eda <manojkiran.eda@gmail.com>

### CODE REVIEW

**Review by Pavithrab7 — APPROVED** [review]

File: host-bmc/host_pdr_handler.hpp [hunk 1]
```diff
@@ -249,16 +249,14 @@ class HostPDRHandler
      *
      *  @param[out] uint16_t    - total table records
```
[the remark this answers is not in the document]
Comment (manojkiraneda) [reply host-bmc/host_pdr_handler.hpp]: done

File: host-bmc/host_pdr_handler.hpp [hunk 2]
```diff
@@ -267,21 +265,19 @@ class HostPDRHandler
      *
```
[the remark this answers is not in the document]
Comment (manojkiraneda) [reply host-bmc/host_pdr_handler.hpp:263]: done

File: host-bmc/host_pdr_handler.cpp [hunk 3]
```diff
@@ -411,6 +412,14 @@ int HostPDRHandler::handleStateSensorEvent(
 
             break;
         }
+        else if (stateSetId[0] == PLDM_STATE_SET_VERSION)
+        {
+            // There is a version changed on any of the dbus objects
+            std::cerr
```
[the remark this answers is not in the document]
Comment (manojkiraneda) [reply host-bmc/host_pdr_handler.cpp]: done

File: host-bmc/host_pdr_handler.cpp [hunk 4]
```diff
@@ -1545,6 +1545,11 @@ void HostPDRHandler::setLocationCode(
                     if (tlv.fruFieldType == PLDM_FRU_FIELD_TYPE_VERSION &&
                         node.entity_type == PLDM_ENTITY_SYSTEM_CHASSIS)
                     {
+                        std::cerr << "Refreshing the mex firmware version : "
```
[the remark this answers is not in the document]
Comment (manojkiraneda) [reply host-bmc/host_pdr_handler.cpp]: done

**Review by jaypadath — APPROVED** [review]

**Pull request merged**